docs/tutorials/upgrading-airbyte.md

```diff
@@ -0,0 +1,89 @@
+_Coming soon! This tutorial is a work in progress for a feature that is currently in development. It is not usable yet!_
+
+# Upgrading Airbyte
+
+## Overview
+
+Airbyte releases a new version of its stable version at least once a week. Some of these releases require updating the data the Airbyte stores internally. This tutorial will describe how to determine if you need to run this ugprade process, and if you do, how to do so. This process does require temporarily turning off Airbyte.
+
+## Determining if you need to Upgrade
+Not every new version will require an upgrade. We follow standard [Semantic Versioning](https://semver.org/) conventions. Data migrations should only be required when upgrading to a new major version of Airbyte. You can always find the latest stable version of Airbyte in our repository [here](https://github.com/airbytehq/airbyte/blob/master/.env#L1). If you are upgrading to a new major version follow the steps below.
+
+If you inadvertently upgrade to a version of Airbyte that is not compatible with your data, the docker containers will not start up and will log an error stating the incompatibility. In these cases, you should downgrade to the previous version that worked and follow the steps below.
+
+## Upgrading (Docker)
+1. Turn off Airbyte
+```
+docker-compose down
+```
+2. Turn on the Airbyte web app, server, and db.
+```
+docker-compose up db server webapp
+```
+3. Navigate to the Admin page in the UI. Then go to the Configuration Tab. Click Export. This will download a gzipped tarball of all of your Airbyte configuration data and sync history. _Note: Any secrets that you have entered into Airbyte will be in this archive, so you should treat it as secret._
+4. Build the Migration App. This is the app that will migrate the data in the archive.
```

Comment:
we should likely publish a docker image to docker hub instead of making a user build locally. i haven't done it yet because it adds some more degrees of freedom in an already ungainly process. would be curious if reviewers would prefer this is done now or not.

the open question for me is what the tag would be on these images. i think they would just be pushed the same way we push the core images, but we would always recommend that the user just use the `:latest` tag. that, and this may be better, they make sure they use the version of docker image that they are trying to upgrade to. the tag would match the version.

Reply:
not fan that we have to build airbyte in order to upgrade. If for any reason it doesn't build, I can't update.

I prefer we release a docker image tagged with the current version. This image has the same lifecycle as the core release

Reply:
okay. i agree. will add it to composebuild, etc.